### PR TITLE
feat:[SpecDecode][Eagle3] Add image-text pair support for Qwen3-VL drafter

### PR DESCRIPTION
**Pull request by shx2005**
## Purpose
Enable single image-text pair support in speculative decoding for Eagle3 (Qwen3-VL), so VL draft models can correctly consume multimodal prefill embeddings.

Example usage:

CUDA_VISIBLE_DEVICES="${CUDA_VISIBLE_DEVICES:-0}" \
```
vllm serve "${BASE_MODEL:-/path/to/qwen3-vl}" \
  --tensor-parallel-size "${TENSOR_PARALLEL_SIZE:-1}" \
  --max-model-len "${MAX_MODEL_LEN:-8192}" \
  --gpu-memory-utilization "${GPU_MEMORY_UTILIZATION:-0.9}" \
  --trust-remote-code \
  --speculative-config "{\"model\":\"${SPEC_CKPT}\",\"method\":\"${SPEC_METHOD:-eagle3}\",\"num_speculative_tokens\":${NUM_SPEC_TOKENS:-3}}" \
  --port "${VLLM_PORT:-8000}" \
  --host "${VLLM_HOST:-127.0.0.1}" \
  --allowed-local-media-path "${ALLOWED_LOCAL_MEDIA_PATH:-/path/to/test/image}"

```

This PR mainly:
- adds a dedicated `Eagle3Qwen3vlForCausalLM` wrapper for VL Eagle3 drafting,
- routes `Eagle3Qwen3vlForCausalLM` to the correct model implementation in registry,

## Test Plan
We re-annotated 4K high-quality image-text samples from `LLaVA-CoT-100K `and trained a compact draft model with `vllm-speculators`, using` Qwen3-VL-8B-Instruct `as the verifier backbone.

## Test Result
On a single H800 PCIe
| Metric | Baseline | Speculative (Eagle3) | Delta |
|---|---:|---:|---:|
| Throughput (token/s) | 149 | 151 | +2 (+1.3%) |

### Speculative Decoding Acceptance (Spec run)

| Metric | Value |
|---|---:|
| Total samples | 22 |
| Total drafted tokens | 60,729 |
| Average drafted tokens | 2,760.41 |
| Weighted per-position acceptance | [0.43, 0.146, 0.04] |
| Conditional acceptance | [0.43, 0.339, 0.277] |

The throughput gain is limited because the draft model was trained on only 4K re-annotated high-quality image-text samples.  
Still, the end-to-end run and acceptance statistics confirm that the VL speculative decoding code path is working correctly.

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan, such as providing test command.
- [x] The test results, such as pasting the results comparison before and after, or e2e results
- [x] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

File: vllm/model_executor/models/qwen3_vl_eagle3.py [hunk 1]
```diff
@@ -0,0 +1,435 @@
...
+                param = params_dict[name]
+                weight_loader = getattr(param, "weight_loader", default_weight_loader)
+                weight_loader(param, loaded_weight)
+            loaded_params.add(name)
+        return loaded_params
+
+
+class Eagle3Qwen3vlForCausalLM(Qwen3ForCausalLM):
+    def __init__(self, *, vllm_config: VllmConfig, prefix: str = ""):
+        nn.Module.__init__(self)
```
[the remark this answers is not in the document]
Comment (shx2005) [reply vllm/model_executor/models/qwen3_vl_eagle3.py:277]: This pattern is pre-existing in Eagle3LlamaForCausalLM as well.
